recursive/recursive_sorting.js

```diff
@@ -23,9 +23,43 @@
 
 
 //helper function that does the partition
-const partition = arr => {}
+const partition = arr => {
+    let pivot = arr[0]
+    let sm_arr = []
+    let lg_arr = []
 
-const quick_sort = arr => {}
+    for(let i = 1; i < arr.length; i++){
+        if(pivot >= arr[i]){
+            sm_arr.unshift(arr[i])
+        }
+        else if(pivot < arr[i]){
+            lg_arr.push(arr[i])
+        }
+    }
+
+    return [sm_arr, [pivot], lg_arr]
```

Comment:
If you returned this as an object rather than array it would make the quick_sort function a lot smaller and a lot quicker because You could quick_sort the sm_arr, lg_arr